llvm/labs/lab2/simonyan_suren/magic_inline.cpp

```diff
@@ -0,0 +1,93 @@
+#include "llvm/ADT/SmallPtrSet.h"
+#include "llvm/IR/BasicBlock.h"
+#include "llvm/IR/Function.h"
+#include "llvm/IR/IRBuilder.h"
+#include "llvm/IR/Instructions.h"
+#include "llvm/Passes/PassBuilder.h"
+#include "llvm/Passes/PassPlugin.h"
+#include "llvm/Transforms/Utils/ValueMapper.h"
+
+namespace {
+
+struct SimonyanInliningPass : public llvm::PassInfoMixin<SimonyanInliningPass> {
+  llvm::PreservedAnalyses run(llvm::Function &F,
+                              llvm::FunctionAnalysisManager &) {
+    llvm::SmallPtrSet<llvm::CallInst *, 8> CallsToRemove;
+
+    for (llvm::BasicBlock &BB : F) {
+      for (llvm::Instruction &Instr : BB) {
+        if (llvm::CallInst *CI = llvm::dyn_cast<llvm::CallInst>(&Instr)) {
+          llvm::Function *Callee = CI->getCalledFunction();
+          if (Callee && Callee->arg_empty() &&
+              Callee->getReturnType()->isVoidTy()) {
+            std::map<llvm::Value *, llvm::Value *> ValueMap;
```

Comment:
There is a special class defined for that `llvm::ValueMap`. Try to use it.

Reply:
Thanks, I'll make changes